src/main/java/org/owasp/html/HtmlPolicyBuilder.java

```diff
@@ -714,6 +735,32 @@ public AttributeBuilder matching(
       });
     }
 
+	/**
+     * Restrict the values allowed by later {@code allow*} calls to those
+     * NOT matching the pattern. This is a convenience method, as inverting
+     * certain patterns can otherwise be quite complex.
+     */
+    public AttributeBuilder notMatching(final Pattern pattern) {
+      return matching(new AttributePolicy() {
+        public @Nullable String apply(String elementName, String attributeName, String value) {
+          return !pattern.matcher(value).matches() ? value : null;
+        }
+      });
+    }
+
+    /**
+     * Restrict the values allowed by later {@code allow*} calls to those
+     * that are not only whitespace or an empty string. Can be combined 
+	 * with calls to {@code matching} or {@code notMatching}.
+     */
+    public AttributeBuilder notEmptyOrWhitespace() {
+      return matching(new AttributePolicy() {
+        public @Nullable String apply(String elementName, String attributeName, String value) {
+          return value.trim().length() > 0 ? value : null;
```

Comment:
I believe that the definition of whitespace in HTML is not the same as the whitespace matched by `String.trim()`.

http://www.w3.org/html/wg/drafts/html/master/single-page.html#space-character says

> The space characters, for the purposes of this specification, are U+0020 SPACE, U+0009 CHARACTER TABULATION (tab), U+000A LINE FEED (LF), U+000C FORM FEED (FF), and U+000D CARRIAGE RETURN (CR).

while [javadoc](http://docs.oracle.com/javase/7/docs/api/java/lang/String.html#trim%28%29) says

> character in the string whose code is greater than '\u0020'